### PR TITLE
Support for directly reading HDF5 and NetCDF MINC files.

### PR DESCRIPTION
**Pull request by rdvincent**
Implement #272. This is still something of a work in progress, but it has reached the point where I believe it will read most MINC 1.0 and MINC 2.0 files correctly. Ideally there should be more testing, enhancement, and documentation before any actual release incorporating this change. But I would like to get comments and review started.

The HDF5 code is large and hairy, which reflects the size and complexity of the specification. HDF5 files incorporate two different classes of objects, and the different classes can coexist in one file. The format is full of interdependencies between different structures, making abstraction a challenge.

In comparison, the NetCDF "classic" format is simple, so the code is much easier to write and read.

### CODE REVIEW

File: src/brainbrowser/volume-viewer/volume-loaders/hdf5.js [hunk 1]
```diff
@@ -0,0 +1,2355 @@
...
+      var value;
+      var n_values;
+      var new_abuf;
+      var i;
+      var spp = dv_offset;
+      if (new_off) {
+        dv_offset = new_off;
+      }
+      switch (typ) {
+      case type_enum.INT8:
```
Comment (natacha-beck) [inline src/brainbrowser/volume-viewer/volume-loaders/hdf5.js]: Nitpick. Can you properly indent the case statement ? Or you indent like that due to jshint ?

File: src/brainbrowser/volume-viewer/volume-loaders/hdf5.js [hunk 2]
```diff
@@ -0,0 +1,2355 @@
...
+      var block_length;
+      if (row <= 1) {
+        block_length = fh.start_blksz;
+      }
+      else {
+        block_length = Math.pow(2, row - 1) * fh.start_blksz;
+      }
+      if (callback) {
+        return callback(row, address, block_offset, block_length);
+      }
```
Comment (natacha-beck) [inline src/brainbrowser/volume-viewer/volume-loaders/hdf5.js]: Some syntax is missing ('{' and '}') no ?

File: src/brainbrowser/volume-viewer/volume-loaders/hdf5.js [hunk 3]
```diff
@@ -0,0 +1,2355 @@
...
+
+  /* Find a dataset with a given name, by recursively searching through
+   * the links. Groups will have 'type' fields of -1, since they contain
+   * no data.
+   * TODO (maybe): Use associative array for children?
+   */
+  function findDataset(link, name, level) {
+    var result;
+    if (link.name === name && link.type > 0) {
+
```
Comment (natacha-beck) [inline src/brainbrowser/volume-viewer/volume-loaders/hdf5.js]: Only need to be print in the console in debug mode, no ?

Comment (natacha-beck) [inline src/brainbrowser/volume-viewer/volume-loaders/hdf5.js]: Only if debug === true ?

File: src/brainbrowser/volume-viewer/volume-loaders/netcdf.js [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+    dimensions();
+    attributes(root);
+    variables(root);
+    records();
+    root.dimensions = file_dimensions;
+    root.numrec = numrec;
+    return root;
+  };
+
+  console.log("NetCDF is loaded!");
```
Comment (natacha-beck) [inline src/brainbrowser/volume-viewer/volume-loaders/netcdf.js]: If debug only, no ?

**Comment by natacha-beck** [comment]
@rdvincent I read all your code. Make some little comments. Nice job, it will be really useful. 
A few general comments: 
- I'm little be lost with code comments, lot of yours function comments does not follow docular rules for comment. I'm not sure which one will have documentation or not, maybe you can take a second look. 
- I saw some console.log without `if (debug)` I don't know if it is what you want.

File: src/brainbrowser/volume-viewer/volume-loaders/hdf5.js (hunk 1 above)
Comment (rdvincent) [inline src/brainbrowser/volume-viewer/volume-loaders/hdf5.js]: I generally follow the common convention of keeping the "case" clauses at the same indentation as the "switch" statement. The brainbrowser codebase is inconsistent here - some of the existing code indents cases, some does not.

File: src/brainbrowser/volume-viewer/volume-loaders/hdf5.js (hunk 2 above)
Comment (rdvincent) [inline src/brainbrowser/volume-viewer/volume-loaders/hdf5.js]: Yes, will fix.

**Comment by rdvincent** [comment]
@natacha-beck I will try to formalize some of the comments to conform to docular. I don't really know the docular standard, but I'll try to follow it where appropriate. Up until now I was mostly just trying to leave comments for myself.

**Pull request merged**